### DISCUSSION TITLE
TypeOrm如何优雅的实现动态表名

### DISCUSSION THREAD

**Discussion by xmsz** [midwayjs/midway#1207] [Q&A]
我现在的表是
```
log_202101
log_202102
...
```

理论上是无限叠加的

所以我想要的是可以动态的指定表名 这样就能解决需求
当然有其他办法肯定更好

然后目前网上的大概是
1. 动态新建数据库连接，这个不合理
2. `repository.metadata.tableName = 'dynamic';`  这种实在有点hack

**Comment by czy88840616** [comment]
这种分表的需求我理解typeorm不适合，直接sql会更好

**Reply by xmsz** [reply]
好吧

**Reply by aaronlam** [reply]
想问下，最后有解决这个问题吗？刚好也碰到这个问题。

**Reply by czy88840616** [reply]
<img width="795" alt="image" src="https://github.com/user-attachments/assets/4b74428d-ce0b-4b9c-8cd6-789f68e66af4" />

**Reply by aaronlam** [reply]
> <img alt="image" width="795" src="https://private-user-images.githubusercontent.com/418820/449562759-4b74428d-ce0b-4b9c-8cd6-789f68e66af4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxMTk2MTMsIm5iZiI6MTc0OTExOTMxMywicGF0aCI6Ii80MTg4MjAvNDQ5NTYyNzU5LTRiNzQ0MjhkLWNlMGItNGI5Yy04Y2Q2LTc4OWY2OGU2NmFmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwNVQxMDI4MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MTEzN2JhMGIwNDNkZTZmYTdjODYyNzI5N2I0Y2ZjZWJiY2Y1YTA5NGNjNGNkNzA4NzRjMGQzOTVmZjYzZDZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.mTd8TRfb6RXCjCFtd-7aAVOJj4ljv0qHaFFW59qdji0">

感谢，用这种方式的确解决了问题，并且实际实操过程中发现需要修改metadata.pathTableName才能生效。仅仅修改metadata.tableName和其他几个涉及tableName的字段都无效。

但是个人思考了一下，这种hack的方式的确是解决了当前的关于分表处理的窘况。但是如果涉及到并发多张分表处理时感觉会出问题，因为修改的是全局的repository实例的metadata。
